mailarchive/mailarchive-subetha/src/java/org/sakaiproject/mailarchive/SakaiMessageHandlerFactory.java

```diff
@@ -349,6 +394,50 @@ public void data(InputStream data) throws RejectException, IOException {
                             log.error("mailarchive General message service exception: (id=" + id + ") (mailId=" + mailId + ") : " + ex, ex);
                         }
                     }
+
+                    // process for each recipient rejected
+                    for (Recipient recipient : rejectedRecipients) {
+
+                        // prepare the message
+                        StringBuilder bodyBuf[] = new StringBuilder[2];
+                        bodyBuf[0] = new StringBuilder();
+                        bodyBuf[1] = new StringBuilder();
+
+                        try {
+                            StringBuilder bodyContentType = new StringBuilder();
+                            parseParts(siteId, msg, id, bodyBuf, bodyContentType, null, -1); // null: dont store incoming attachments as sakai resources, or otherwise process
+
+                            if (bodyContentType.length() > 0) {
+                                // save the content type of the message body - which may be different from the overall MIME type of the message (multipart, etc)
+                                mailHeaders.add(HEADER_INNER_CONTENT_TYPE + ": " + bodyContentType);
+                            }
+                        } catch (MessagingException e) {
+                            // NOTE: if this happens it just means we don't get the extra header, not the end of the world
+                            log.warn("MessagingException: service(): msg.getContent() threw: {}", e);
+                        } catch (IOException e) {
+                            // NOTE: if this happens it just means we don't get the extra header, not the end of the world
+                            log.warn("IOException: service(): msg.getContent() threw: {}", e);
+                        }
+
+                        List<Attachment> outgoingAttachments = new LinkedList<>();
+                        outgoingAttachments.add(new Attachment(new ByteArrayDataSource(bodyBuf[0].toString(), "plain/text")));
+                        outgoingAttachments.add(new Attachment(new ByteArrayDataSource(bodyBuf[1].toString(), "html/text")));
+
+                        String errMsg = recipient.errorExplanation + "\n\n";
+                        String mailSupport = StringUtils.trimToNull(serverConfigurationService.getString("mail.support"));
+                        if (mailSupport != null) {
+                            errMsg += rb.getFormattedMessage("err_questions", mailSupport) + "\n";
+                        }
+
+                        try {
+                            InternetAddress errFrom = new InternetAddress("postmaster@".concat(serverConfigurationService.getServerName()));
```

Comment:
Should this be using smtpFrom@org.sakaiproject.email.api.EmailService instead?

Reply:
Possibly; I'll look into this.